lib/ts/recipe/multifactorauth/types.ts

```diff
@@ -135,6 +135,65 @@ export type APIInterface = {
                 }
               | GeneralErrorResponse
           >);
+
+    factorsSetupForUserGET?: (input: {
+        session: SessionContainerInterface;
+        options: APIOptions;
+        userContext: UserContext;
+    }) => Promise<
+        | {
+              status: "OK";
+              factors: {
+                  id: string;
+                  details: JSONObject;
+              }[];
+              backupCodes: string[] | undefined;
+          }
+        | GeneralErrorResponse
+    >;
+
+    addFactorForUserPOST?: (input: {
+        factor: {
+            id: string;
+        };
+        session: SessionContainerInterface;
+        options: APIOptions;
+        userContext: UserContext;
+    }) => Promise<
+        | {
+              status: "OK";
+              backupCodes: string[];
+          }
+        | GeneralErrorResponse
+    >;
+
+    deleteFactorForUserDELETE?: (input: {
+        factorId: string;
+        session: SessionContainerInterface;
+        options: APIOptions;
+        userContext: UserContext;
+    }) => Promise<
+        | {
+              status: "OK";
+          }
+        | GeneralErrorResponse
+    >;
+
```

Comment:
add api to get factors for a user

Reply:
The way this would work is to query what factors are enabled for the user in the metadata + union it with factors that are completed in the current session.